ui-react/packages/atlasmap-core/src/services/mapping-management.service.ts

```diff
@@ -425,9 +425,11 @@ export class MappingManagementService {
     if (selectedField != null) {
       mapping.addField(selectedField, false);
       this.updateTransition(mapping);
+      this.selectMapping(mapping);
       this.notifyMappingUpdated();
+    } else {
+      this.selectMapping(mapping);
```

Comment:
Is this change really necessary?  Can't the selectMapping occur at the end?

Reply:
Yes, AFAICT, because the selectMapping call is what sets the activeMapping, which the subsequent notifyMappingUpdated call relies upon to update the list of mappings.